### ISSUE TITLE
Support ::ng-deep as an alternative to /deep/

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular#454]
Angular will need some replacement
sass/dart-sass#154

**Comment by matanlurey** [comment]
We have a number of ideas, but nothing I'm able to publish yet. Will update soon.

**Comment by zoechi** [comment]
That's what I wanted to hear, that it's on your radar. No need to provide details right now.

**Comment by matanlurey** [comment]
We've decided to deprecate `/deep/` and `>>>`, and support `:ng-deep` as a drop-in.

Here is our internal test case for this feature:

```dart
  test('should handle ::ng-deep', () {
    var css = '::ng-deep y {}';
    shimAndExpect(css, 'y {}');
    css = 'x ::ng-deep y {}';
    shimAndExpect(css, 'x.$content y {}');
    css = ':host > ::ng-deep .x {}';
    shimAndExpect(css, '.$host > .x {}');
    css = ':host ::ng-deep > .x {}';
    shimAndExpect(css, '.$host > .x {}');
    css = ':host > ::ng-deep > .x {}';
    shimAndExpect(css, '.$host > > .x {}');
  });
```

We've gotten confirmation this will be supported (parsable) in all major SASS tooling.

AngularDart `4.x.x` will be the last major release to support either `/deep`/ or `>>>` (both, including `::ng-deep`, will be available in 4.0). As a result, _native_ shadow DOM will also be deprecated (but not immediately removed), since there is no way after [Chrome's `/deep/` deprecation lands](https://www.chromestatus.com/features/4964279606312960), and we don't have another story (yet).

**Comment by alorenzen** [comment]
@leonsenft Was this included as part of  #471 ?

**Comment by leonsenft** [comment]
@alorenzen Yes it was. Looks like it got swallowed up by https://github.com/dart-lang/angular/commit/302c3865c68a7ed5277a1804aea772be4b0f0df6 and we lost the commit message.
